h/models/group.py

```diff
@@ -98,6 +99,18 @@ def groupid(self):
             authority_provided_id=self.authority_provided_id,
             authority=self.authority)
 
+    @groupid.setter
+    def groupid(self, value):
+        """
+        Deconstruct a formatted ``groupid`` and set its constituent properties
+        on the instance.
+
+        :raises ValueError: if ``groupid`` is an invalid format
+        """
+        groupid_parts = split_groupid(value)
+        self.authority_provided_id = groupid_parts['authority_provided_id']
+        self.authority = groupid_parts['authority']
```

Comment:
Someone could already edit `Group.authority` directly, so I guess the danger here is someone editing `groupid` and not realising that may change the group's authority.

If you want to prevent the authority of existing groups from ever being altered, you could have this method raise of `self.authority` is not `None` when this method is called. Similarly `self.authority_provided_id` could raise if already set. Of course both fields can already be updated currently by setting them directly instead of via `groupid`, so you'd have to prevent that too. I think it's probably not worth trying to prevent that at the model level

Reply:
Or raise `if self.authority not in (None, groupid_parts["authority"])`, that would allow `authority_provided_id` to be changed via `groupid` but not allow `authority` to.

Reply:
I think this is safe though as long as we trust the rest of our code not to flinging new groupids at models willy nilly